src/odemis/driver/technolution.py

```diff
@@ -241,8 +443,111 @@ def __init__(self, name, role, parent, **kwargs):
+                                                        dwelltime * cell_size[0], scan_period))))
+
+        flyback_points = 0 * numpy.arange(0, self.physicalFlybackTime, scan_period)
+        setpoints_row = numpy.tile(numpy.concatenate((acq_delay_points, rise_points, flyback_points)), cell_size[1])
+        base = self.scanOffset.value[0] - 0.5 * cell_size[0] * scan_gain[0]
+        setpoints = base + numpy.concatenate((scan_delay_points, setpoints_row))
+
+        if setpoints.min() < -32768 or setpoints.max() > 32767:
+            raise ValueError("Setpoint values are to big/small to be handled by the ASM API")
+        return setpoints.astype(int).tolist()
+
+    def _getYAcqSetpoints(self):
+        scan_period = self.clockPeriod.value
+        scan_gain = self.scanGain.value
+        dwelltime = self.parent._ebeam_scanner.dwellTime.value  # s
+        scan_delay = self.parent._ebeam_scanner.scanDelay.value  # s
+        cell_size = self.parent._mppc.cellCompleteResolution.value  # pixels
+        acq_delay = self.parent._mppc.acqDelay.value - self.parent._ebeam_scanner.scanDelay.value[0]
+
+        # Function to find off beatness of the setpoint lines
+        # "(scan_period - input % scan_period) % scan_period" because when input is a multiple of scan_period
+        # the outcome should be 0, other wise the outcome could be the time left in that scan_period.
+        off_beat = lambda input: (scan_period - input % scan_period) % scan_period
+
+        # all units in seconds
+        #TODO K.K. if testcases in place change arange to zeros
+        scan_delay_points = 0 * numpy.arange(0, scan_delay[1], scan_period)
+        acq_delay_points = 0 * numpy.arange(off_beat(scan_delay[1]), acq_delay, scan_period)
+
+        rise_points = 0 * numpy.arange(off_beat(scan_delay[1] + acq_delay),
+                                                            dwelltime * cell_size[0], scan_period)
+
+        #TODO K.K. remove part compairing and checking
+        rise_points1 = 0 * numpy.array(list(map(lambda t: (scan_gain[1] / dwelltime) * t,
+                                               numpy.arange(off_beat(scan_delay[1] + acq_delay),
+                                                            dwelltime * cell_size[0], scan_period))))
+
+        if not numpy.array_equal(rise_points1, rise_points):
+            raise ValueError
+
+        flyback_points = 0 * numpy.arange(0, self.physicalFlybackTime, scan_period)
+        setpoints_row = numpy.concatenate((acq_delay_points, rise_points, flyback_points))
+        setpoints = numpy.array([scan_gain[1] * i + setpoints_row for i in range(0, cell_size[1])]).reshape(-1)
+        base = self.scanOffset.value[1] - 0.5 * cell_size[0] * scan_gain[1]
+        setpoints = base + numpy.concatenate((scan_delay_points, setpoints))
+
+        if setpoints.min() < -32768 or setpoints.max() > 32767:
+            raise ValueError("Setpoint values are to big/small to be handeled by the ASM API")
```

Comment:
`Acquisition setpoints for the descanner are out of range as excepted by the ASM API`